cli/cmd/inject_test.go

```diff
@@ -310,6 +310,28 @@ func TestUninjectAndInject(t *testing.T) {
 			injectProxy:      true,
 			testInjectConfig: ingressConfig,
 		},
+		{
+			inputFileName:  "inject_emojivoto_deployment.input.yml",
+			goldenFileName: "inject_emojivoto_deployment_default_inbound_policy.golden.yml",
+			reportFileName: "inject_emojivoto_deployment_default_inbound_policy.golden.report",
+			injectProxy:    false,
+			testInjectConfig: func() *linkerd2.Values {
+				values := defaultConfig()
+				values.Proxy.DefaultInboundPolicy = k8s.AllAuthenticated
+				return values
+			}(),
+		},
+		{
+			inputFileName:  "inject_emojivoto_pod.input.yml",
+			goldenFileName: "inject_emojivoto_pod_default_inbound_policy.golden.yml",
+			reportFileName: "inject_emojivoto_pod_default_inbound_policy.golden.report",
+			injectProxy:    false,
+			testInjectConfig: func() *linkerd2.Values {
+				values := defaultConfig()
+				values.Proxy.DefaultInboundPolicy = "all-authenticated"
```

Comment:
since k8s.AllAuthenticated = "all-authenticated", aren't these test cases identical?

Reply:
Aside from not using the constant, the difference between the two tests is that one is injecting into a deploy spec and the other into a pod spec.

Reply:
It's about as identical as these two existing test cases.

https://github.com/linkerd/linkerd2/blob/ccf38bd71bd39b800e800cba0de67574050c267d/cli/cmd/inject_test.go#L107-L113

https://github.com/linkerd/linkerd2/blob/ccf38bd71bd39b800e800cba0de67574050c267d/cli/cmd/inject_test.go#L192-L198